### ISSUE TITLE
Enhancement: Pet Clinic reference pipeline should create/dispose CI environment

### ISSUE THREAD

**Issue by deors** [Accenture/adop-cartridge-java#7]
Enhance Pet Clinic reference pipeline, so it creates and disposes the CI environment during the process. The environment name will be named as CI-${B} to allow for multiple parallel job execution. The workflow will be:
1- Compile/package
2- Unit tests
3- Static code analysis
4- Create environment
5- Deploy package
6- Regression tests
7- Destroy environment

**Comment by nickdgriffin** [comment]
The cartridge now automatically creates the CI environment when it is loaded by calling the job, which we could remove if this was implemented.

**Comment by deors** [comment]
Yeah. The idea I had is that instead of creating one and leaving it, we could have for each pipeline execution, a new DEV-type environment created with build number in it, and disposed after tests are executed. This way, we could for example have multiple jobs in parallel testing different features, and we could be potentially saving resources as we don't maintain running an idle environment.
In the future, we may follow a similar approach for deploys to PROD, following a blue/green approach, although that is not so high in my priority list, as I don't foresee blue/green deployments as frequent.
